diff -ruN /original-sources/.gradle/buildOutputCleanup/cache.properties ./.gradle/buildOutputCleanup/cache.properties
--- /original-sources/.gradle/buildOutputCleanup/cache.properties	1970-01-01 00:00:00.000000000 +0000
+++ ./.gradle/buildOutputCleanup/cache.properties	2023-10-23 05:22:00.950740722 +0000
@@ -0,0 +1,2 @@
+#Fri Jul 28 11:00:26 GMT 2023
+gradle.version=7.5.1
diff -ruN /original-sources/buildSrc/.gradle/buildOutputCleanup/cache.properties ./buildSrc/.gradle/buildOutputCleanup/cache.properties
--- /original-sources/buildSrc/.gradle/buildOutputCleanup/cache.properties	1970-01-01 00:00:00.000000000 +0000
+++ ./buildSrc/.gradle/buildOutputCleanup/cache.properties	2023-10-23 05:22:00.896739395 +0000
@@ -0,0 +1,2 @@
+#Fri Jul 28 10:59:40 GMT 2023
+gradle.version=7.5.1
diff -ruN /original-sources/buildSrc/build/kotlin/buildSrcjar-classes.txt ./buildSrc/build/kotlin/buildSrcjar-classes.txt
--- /original-sources/buildSrc/build/kotlin/buildSrcjar-classes.txt	1970-01-01 00:00:00.000000000 +0000
+++ ./buildSrc/build/kotlin/buildSrcjar-classes.txt	2023-10-23 05:22:00.896739395 +0000
@@ -0,0 +1 @@
+/project/sources/buildSrc/build/classes/kotlin/main/org/postgresql/buildtools/JavaCommentPreprocessorTask$inputDirectories$1$1.class:/project/sources/buildSrc/build/classes/kotlin/main/org/postgresql/buildtools/JavaCommentPreprocessorTask$run$$inlined$apply$lambda$1.class:/project/sources/buildSrc/build/classes/kotlin/main/org/postgresql/buildtools/JavaCommentPreprocessorTask.class
\ No newline at end of file
diff -ruN /original-sources/buildSrc/build/kotlin/compileKotlin/cacheable/caches-jvm/lookups/counters.tab ./buildSrc/build/kotlin/compileKotlin/cacheable/caches-jvm/lookups/counters.tab
--- /original-sources/buildSrc/build/kotlin/compileKotlin/cacheable/caches-jvm/lookups/counters.tab	1970-01-01 00:00:00.000000000 +0000
+++ ./buildSrc/build/kotlin/compileKotlin/cacheable/caches-jvm/lookups/counters.tab	2023-10-23 05:22:00.897739420 +0000
@@ -0,0 +1,2 @@
+1
+0
\ No newline at end of file
diff -ruN /original-sources/buildSrc/build/pluginUnderTestMetadata/plugin-under-test-metadata.properties ./buildSrc/build/pluginUnderTestMetadata/plugin-under-test-metadata.properties
--- /original-sources/buildSrc/build/pluginUnderTestMetadata/plugin-under-test-metadata.properties	1970-01-01 00:00:00.000000000 +0000
+++ ./buildSrc/build/pluginUnderTestMetadata/plugin-under-test-metadata.properties	2023-10-23 05:22:00.897739420 +0000
@@ -0,0 +1 @@
+implementation-classpath=/project/sources/buildSrc/build/classes/java/main\:/project/sources/buildSrc/build/classes/groovy/main\:/project/sources/buildSrc/build/classes/kotlin/main\:/project/sources/buildSrc/build/resources/main\:/root/.gradle/caches/modules-2/files-2.1/com.igormaznitsa/jcp/7.0.2/ba17268f37d8b1d13222000686a42527423cc078/jcp-7.0.2.jar
diff -ruN /original-sources/buildSrc/build/tmp/jar/MANIFEST.MF ./buildSrc/build/tmp/jar/MANIFEST.MF
--- /original-sources/buildSrc/build/tmp/jar/MANIFEST.MF	1970-01-01 00:00:00.000000000 +0000
+++ ./buildSrc/build/tmp/jar/MANIFEST.MF	2023-10-23 05:22:00.897739420 +0000
@@ -0,0 +1,2 @@
+Manifest-Version: 1.0
+
diff -ruN /original-sources/pgjdbc/src/test/java/org/postgresql/jdbc/PgSQLXMLTest.java ./pgjdbc/src/test/java/org/postgresql/jdbc/PgSQLXMLTest.java
--- /original-sources/pgjdbc/src/test/java/org/postgresql/jdbc/PgSQLXMLTest.java	2023-02-16 14:21:18.000000000 +0000
+++ ./pgjdbc/src/test/java/org/postgresql/jdbc/PgSQLXMLTest.java	2023-10-23 08:46:14.919949947 +0000
@@ -45,7 +45,7 @@
   @Before
   public void setUp() throws Exception {
     super.setUp();
-    TestUtil.createTempTable(con, "xmltab", "x xml");
+    TestUtil.createTempTable(con, "xmltab", "x xml, _ydb_stub_id SERIAL PRIMARY KEY");
   }
 
   @Test
diff -ruN /original-sources/pgjdbc/src/test/java/org/postgresql/test/TestUtil.java ./pgjdbc/src/test/java/org/postgresql/test/TestUtil.java
--- /original-sources/pgjdbc/src/test/java/org/postgresql/test/TestUtil.java	2023-02-16 14:21:18.000000000 +0000
+++ ./pgjdbc/src/test/java/org/postgresql/test/TestUtil.java	2023-10-23 05:22:00.948740673 +0000
@@ -455,7 +455,7 @@
       dropTable(con, table);
 
       // Now create the table
-      st.executeUpdate("create temp table " + table + " (" + columns + ")");
+      st.executeUpdate("create table " + table + " (" + columns + ")");
     } finally {
       closeQuietly(st);
     }
@@ -640,10 +640,10 @@
     try {
       if (con.getAutoCommit()) {
         // Not in a transaction so ignore error for missing object
-        stmt.executeUpdate("DROP " + type + " IF EXISTS " + name + " CASCADE");
+        stmt.executeUpdate("DROP " + type + " IF EXISTS " + name);
       } else {
         // In a transaction so do not ignore errors for missing object
-        stmt.executeUpdate("DROP " + type + " " + name + " CASCADE");
+        stmt.executeUpdate("DROP " + type + " " + name);
       }
     } finally {
       closeQuietly(stmt);
diff -ruN /original-sources/pgjdbc/src/test/java/org/postgresql/test/jdbc2/StatementTest.java ./pgjdbc/src/test/java/org/postgresql/test/jdbc2/StatementTest.java
--- /original-sources/pgjdbc/src/test/java/org/postgresql/test/jdbc2/StatementTest.java	2023-02-16 14:21:18.000000000 +0000
+++ ./pgjdbc/src/test/java/org/postgresql/test/jdbc2/StatementTest.java	2023-10-23 05:22:00.938740427 +0000
@@ -57,11 +57,11 @@
   @Before
   public void setUp() throws Exception {
     con = TestUtil.openDB();
-    TestUtil.createTempTable(con, "test_statement", "i int");
+    TestUtil.createTempTable(con, "test_statement", "i int, _auto_id SERIAL PRIMARY KEY");
     TestUtil.createTempTable(con, "escapetest",
-        "ts timestamp, d date, t time, \")\" varchar(5), \"\"\"){a}'\" text ");
-    TestUtil.createTempTable(con, "comparisontest", "str1 varchar(5), str2 varchar(15)");
-    TestUtil.createTable(con, "test_lock", "name text");
+        "ts timestamp, d date, t time, _auto_id SERIAL PRIMARY KEY, \")\" varchar(5), \"\"\"){a}'\" text ");
+    TestUtil.createTempTable(con, "comparisontest", "str1 varchar(5), str2 varchar(15), _auto_id SERIAL PRIMARY KEY");
+    TestUtil.createTable(con, "test_lock", "name text, _auto_id SERIAL PRIMARY KEY");
     Statement stmt = con.createStatement();
     stmt.executeUpdate(TestUtil.insertSQL("comparisontest", "str1,str2", "'_abcd','_found'"));
     stmt.executeUpdate(TestUtil.insertSQL("comparisontest", "str1,str2", "'%abcd','%found'"));
@@ -74,8 +74,8 @@
     TestUtil.dropTable(con, "escapetest");
     TestUtil.dropTable(con, "comparisontest");
     TestUtil.dropTable(con, "test_lock");
-    TestUtil.execute("DROP FUNCTION IF EXISTS notify_loop()",con);
-    TestUtil.execute("DROP FUNCTION IF EXISTS notify_then_sleep()",con);
+    // TestUtil.execute("DROP FUNCTION IF EXISTS notify_loop()",con);
+    // TestUtil.execute("DROP FUNCTION IF EXISTS notify_then_sleep()",con);
     con.close();
   }
 
@@ -163,7 +163,7 @@
     count = stmt.executeUpdate("UPDATE test_statement SET i=4");
     assertEquals(2, count);
 
-    count = stmt.executeUpdate("CREATE TEMP TABLE another_table (a int)");
+    count = stmt.executeUpdate("CREATE TEMP TABLE another_table (a int, _auto_id SERIAL PRIMARY KEY)");
     assertEquals(0, count);
 
     if (TestUtil.haveMinimumServerVersion(con, ServerVersion.v9_0)) {
@@ -603,25 +603,6 @@
     }
   }
 
-  /**
-   * The parser tries to break multiple statements into individual queries as required by the V3
-   * extended query protocol. It can be a little overzealous sometimes and this test ensures we keep
-   * multiple rule actions together in one statement.
-   */
-  @Test
-  public void testParsingSemiColons() throws SQLException {
-    Statement stmt = con.createStatement();
-    stmt.execute(
-        "CREATE RULE r1 AS ON INSERT TO escapetest DO (DELETE FROM test_statement ; INSERT INTO test_statement VALUES (1); INSERT INTO test_statement VALUES (2); );");
-    stmt.executeUpdate("INSERT INTO escapetest(ts) VALUES (NULL)");
-    ResultSet rs = stmt.executeQuery("SELECT i from test_statement ORDER BY i");
-    assertTrue(rs.next());
-    assertEquals(1, rs.getInt(1));
-    assertTrue(rs.next());
-    assertEquals(2, rs.getInt(1));
-    assertTrue(!rs.next());
-  }
-
   @Test
   public void testParsingDollarQuotes() throws SQLException {
     // dollar-quotes are supported in the backend since version 8.0
