diff -ruN /original-sources/conn_test.go ./conn_test.go
--- /original-sources/conn_test.go	2023-04-26 04:34:24.000000000 +0000
+++ ./conn_test.go	2023-09-15 07:57:44.972012573 +0000
@@ -318,7 +318,7 @@
 
 	if !r1.Next() {
 		if r.Err() != nil {
-			t.Fatal(r1.Err())
+			t.Fatal(r.Err())
 		}
 		t.Fatal("expected row")
 	}
@@ -973,7 +973,7 @@
 	db := openTestConn(t)
 	defer db.Close()
 
-	_, err := db.Exec("create temp table test (i integer)")
+	_, err := db.Exec("create temp table test (i integer, _stub_id Serial PRIMARY KEY)")
 	if err != nil {
 		t.Fatal(err)
 	}
@@ -1290,7 +1290,7 @@
 
 	if !r.Next() {
 		if r.Err() != nil {
-			t.Fatal(err)
+			t.Fatal(r.Err())
 		}
 		t.Fatal("expected row")
 	}
@@ -1305,7 +1305,7 @@
 
 	if !r.Next() {
 		if r.Err() != nil {
-			t.Fatal(err)
+			t.Fatal(r.Err())
 		}
 		t.Fatal("expected row")
 	}
@@ -1351,11 +1351,11 @@
 	db := openTestConn(t)
 	defer db.Close()
 
-	_, err := db.Exec("CREATE TEMP TABLE temp (a int)")
+	_, err := db.Exec("CREATE TEMP TABLE temp (a int primary key)")
 	if err != nil {
 		t.Fatal(err)
 	}
-	sqlInsert := "INSERT INTO temp VALUES (1)"
+	sqlInsert := "INSERT INTO temp (a) VALUES (1)"
 	sqlSelect := "SELECT * FROM temp"
 	tx, err := db.Begin()
 	if err != nil {
@@ -1501,7 +1501,7 @@
 		}
 
 		value, success := tryGetParameterValue()
-		if success != test.success && !test.success {
+		if success != test.success && !success {
 			t.Fatalf("%v: unexpected error: %v", test.conninfo, err)
 		}
 		if success != test.success {
diff -ruN /original-sources/issues_test.go ./issues_test.go
--- /original-sources/issues_test.go	2023-04-26 04:34:24.000000000 +0000
+++ ./issues_test.go	2023-08-22 09:35:23.189760257 +0000
@@ -113,7 +113,7 @@
 		time.Sleep(10 * time.Millisecond)
 		cancel()
 	}()
-	row := db.QueryRowContext(ctx, "select pg_sleep(0.5)")
+	row := db.QueryRowContext(ctx, "select pg_sleep(4::float8)")
 	var pgSleepVoid string
 	err := row.Scan(&pgSleepVoid)
 	if pgErr := (*Error)(nil); !(errors.As(err, &pgErr) && pgErr.Code == cancelErrorCode) {
